src/envs/screws.py

```diff
@@ -0,0 +1,509 @@
+
+        # Static objects (always exist no matter the settings).
+        self.robot = Object("robby", self._gripper_type)
+        self._receptacle = Object("receptacle", self._receptacle_type)
+
+    def _generate_train_tasks(self) -> List[Task]:
+        return self._get_tasks(num_tasks=CFG.num_train_tasks,
+                               possible_num_screws=self.num_screws_train,
+                               rng=self._train_rng)
+
+    def _generate_test_tasks(self) -> List[Task]:
+        return self._get_tasks(num_tasks=CFG.num_test_tasks,
+                               possible_num_screws=self.num_screws_test,
+                               rng=self._test_rng)
+
+    @classmethod
+    def get_name(cls) -> str:
+        return "screws"
+
+    def render_state(self,
+                     state: State,
+                     task: Task,
+                     action: Optional[Action] = None,
+                     caption: Optional[str] = None) -> List[Image]:
+        fig, ax = plt.subplots(1, 1)
+        plt.xlim([
+            self.rz_x_lb - 2 * self._screw_width,
+            self.rz_x_ub + 2 * self._screw_width
+        ])
+        plt.ylim([self.rz_y_lb, self.rz_y_ub])
+
+        # Draw receptacle
+        receptacle_x = state.get(self._receptacle, "pose_x")
+        receptacle_y = state.get(self._receptacle, "pose_y")
+        receptacle_width = state.get(self._receptacle, "width")
+        plt.plot([
+            receptacle_x - receptacle_width / 2.0,
+            receptacle_x + receptacle_width / 2.0
+        ], [receptacle_y, receptacle_y],
+                 color="green")
+
+        # Draw the gripper
+        gripper_x = state.get(self.robot, "pose_x")
+        gripper_y = state.get(self.robot, "pose_y")
+        gripper_width = state.get(self.robot, "width")
+        # NOTE: this 1.3 factor is just to make the gripper wide enough so that
+        # even when it grabs a block on its very edge, it doesn't look strange
+        # in the visualization.
+        plt.plot(
+            [(gripper_x - gripper_width / 2.0) + (self._screw_width * 1.3),
+             (gripper_x + gripper_width / 2.0) + (self._screw_width * 1.3)],
+            [gripper_y, gripper_y],
+            color="black")
+        plt.plot(
+            [(gripper_x - gripper_width / 2.0) + (self._screw_width * 1.3),
+             (gripper_x - gripper_width / 2.0) + (self._screw_width * 1.3)],
+            [gripper_y - gripper_width / 4.0, gripper_y],
+            color="black")
+        plt.plot(
+            [(gripper_x + gripper_width / 2.0) + (self._screw_width * 1.3),
+             (gripper_x + gripper_width / 2.0) + (self._screw_width * 1.3)],
+            [gripper_y - gripper_width / 4.0, gripper_y],
+            color="black")
+
+        # Find which object is the goal screw.
+        assert len(task.goal) == 1
+        goal_pred = list(task.goal)[0]
+        goal_screw = goal_pred.objects[0]
+        assert goal_screw.type.name == 'screw'
```

Comment:
can probably just remove this assertion, but if you want, you can do `goal_screw.is_instance(self._screw_type)` instead